### PR TITLE
Jointable Queries

### PR DESCRIPTION
**Pull request by gabyse1**
# Changes to the project 😃 🆕 
- Add vets, spetializations, and visits tables.
- Insert data into vets, spetializatins, and visits tables.
- Make queries using join tables or intersection entities: 
  - Who was the last animal seen by William Tatcher?
  - How many different animals did Stephanie Mendez see?
  - List all vets and their specialties, including vets with no specialties.
  - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
  - What animal has the most visits to vets?
  - Who was Maisy Smith's first visit?
  - Details for most recent visit: animal information, vet information, and date of visit.
  - How many visits were with a vet that did not specialize in that animal's species?
  - What specialty should Maisy Smith consider getting? Look for the species she gets the most.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -175,3 +175,115 @@ HAVING COUNT(owners.full_name) = (SELECT MAX(NUMPETS)
...
+SELECT animals.name AS ANIMAL, visits.date_of_visit, vets.name AS VET
+FROM animals INNER JOIN visits ON animals.id = visits.animal_id
+INNER JOIN vets ON visits.vet_id = vets.id
+WHERE vets.name = 'William Tatcher' AND visits.date_of_visit = (SELECT MAX(visits.date_of_visit)
+                            FROM vets INNER JOIN visits ON vets.id = visits.vet_id
+                            WHERE vets.name = 'William Tatcher');
+ animal  | date_of_visit |       vet
+---------+---------------+-----------------
+ Blossom | 2021-01-11    | William Tatcher
+(1 row)
```
Comment (rukundoeric) [inline queries.sql:191]: - [ ] <b>OPTIONAL:</b> Kindly always remember to comment out the results.

File: queries.sql [hunk 2]
```diff
@@ -175,3 +175,115 @@ HAVING COUNT(owners.full_name) = (SELECT MAX(NUMPETS)
...
+
+SELECT vets.name AS VET, COUNT(DISTINCT visits.animal_id) AS NUMANIMALS
+FROM visits INNER JOIN vets ON visits.vet_id = vets.id
+GROUP BY (vets.name)
+HAVING vets.name = 'Stephanie Mendez';
+       vet        | numanimals
+------------------+------------
+ Stephanie Mendez |          4
+(1 row)
```
Comment (rukundoeric) [inline queries.sql:200]: - [ ] <b>OPTIONAL:</b> The same thing goes here. Kindly always remember to comment out the results.

File: queries.sql [hunk 3]
```diff
@@ -175,3 +175,115 @@ HAVING COUNT(owners.full_name) = (SELECT MAX(NUMPETS)
...
+FROM vets LEFT JOIN spetializations ON vets.id = spetializations.vet_id
+LEFT JOIN species ON spetializations.specie_id = species.id;
+       vets       | spetializations
+------------------+-----------------
+ William Tatcher  | Pokemon
+ Stephanie Mendez | Digimon
+ Stephanie Mendez | Pokemon
+ Jack Harkness    | Digimon
+ Maisy Smith      |
+(5 rows)
```
Comment (rukundoeric) [inline queries.sql:212]: - [ ] <b>OPTIONAL:</b> Here too.

File: queries.sql [hunk 4]
```diff
@@ -175,3 +175,115 @@ HAVING COUNT(owners.full_name) = (SELECT MAX(NUMPETS)
...
+      HAVING COUNT(animals.specie_id) = (SELECT MAX(VISITSPERSPECIE)
+                                        FROM (SELECT animals.specie_id, count(animals.specie_id) as VISITSPERSPECIE
+                                              FROM (((SELECT * FROM vets WHERE vets.name = 'Maisy Smith') VET INNER JOIN visits ON VET.id = visits.vet_id)
+                                              INNER JOIN animals ON visits.animal_id = animals.id)
+                                              GROUP BY animals.specie_id) VETVISITS)
+    ) MOSTSPECIE INNER JOIN species ON MOSTSPECIE.specie_id = species.id;
+ likelyspetialization
+----------------
+ Digimon
+(1 row)
```
Comment (rukundoeric) [inline queries.sql:289]: - [ ] <b>OPTIONAL:</b> And all of those query result too.

**Review by rukundoeric — APPROVED** [review]
Hi @gabyse1,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - [ ] check the comments under review

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------

File: queries.sql (hunk 4 above)
Comment (gabyse1) [reply queries.sql:289]: # Hi @rukundoeric 😃 
Thank you for the review. I will commet them.
Greetings. 👋

**Pull request merged**